### PR TITLE
Add blog Cadence bad practice 1

### PR DESCRIPTION
**Pull request by chopincode**

### CODE REVIEW

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+title: Bad practices and Anti-patterns with Cadence (Part 1)
+
+date: 2023-07-10
+author: Chris Qin
+authorlink: https://www.linkedin.com/in/chrisqin0610/
+---
+
+In the upcoming blog series, we will delve into a discussion about common bad practices and anti-patterns related to Cadence. As diverse teams often encounter distinct business use cases, it becomes imperative to address the most frequently reported issues in Cadence workflows. To provide valuable insights and guidance, the Cadence team has meticulously compiled these common challenges based on customer feedback.
+
+* Reuse same workflow ID for very active/continuous running workflows
```
Comment (sharanf) [inline src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:11]: This sounds a little odd. 'Reusing the same workflow ID ....'  would work better

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+At first glance, the practice of maximizing workflow code reusability may seem like a good idea. However, it can cause significant performance problems. This is because Cadence organizes workflows based on their unique IDs, using a process called partitioning. If a workflow receives a large number of updates in a short period or frequently starts new runs using the `continueAsNew` function, all these updates will be directed to the same shard. Unfortunately, the Cadence backend is not equipped to handle this concentrated workload efficiently. As a result, a situation known as a "hot shard" arises, overloading the Cadence backend and worsening the problem.
```
Comment (sharanf) [inline src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:13]: You might want to highlight the word 'partitioning' in bold so it stands out as it is a specific term.
Also 'a short period' is okay but you could be more specific by making it 'a short period of time'

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Solution:
+Well, the best way to avoid this is simply just design your workflow in the way such that each workflow owns a uniformly distributed workflow ID across your Cadence domain. This will make sure that Cadence backend is able to evenly distribute the traffic with proper partition on your workflowIDs.
+
+* Excessive batch jobs or enormous number of timers triggered at the same time
```
Comment (sharanf) [inline src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:18]: I've not seen enormous used like this but it's probably OK - needs an in front e.g 'an enormous.....'

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Cadence has the capability to handle a large number of concurrent tasks initiated simultaneously, but tampering with this feature can lead to issues within the Cadence system. Consider a scenario where millions of jobs are scheduled to start at the same time and are expected to finish within a specific time interval. Cadence faces the challenge of understanding the desired behavior of customers in such cases. It is uncertain whether the intention is to complete all jobs simultaneously, provide progressive updates in parallel, or finish all jobs before a given deadline. This ambiguity arises due to the independent nature of each job and the difficulty in predicting their outcomes.
+
+Moreover, Cadence workers utilize a sticky cache by default to optimize the runtime of workflows. However, when an overwhelming number of parallel workflows cannot fit into the cache, it can result in cache thrashing. This, in turn, leads to a quadratic increase in runtime complexity, specifically O(n^2), exacerbating the overall performance of the system.
```
Comment (sharanf) [inline src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:22]: Maybe highlight the phrase 'cache thrashing' as it is a specific term we are defining.

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md (hunk 1 above)
Comment (chopincode) [reply src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:11]: Done

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md (hunk 2 above)
Comment (chopincode) [reply src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:13]: done

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md (hunk 3 above)
Comment (chopincode) [reply src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md:18]: done

File: src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md (hunk 4 above)
Comment (chopincode) [reply src/blog/_posts/2023-07-10-cadence-bad-practices-part-1.md]: done

**Review by allenchen2244 — APPROVED** [review]

**Pull request merged**